### PR TITLE
Adding credentials set support ms graph mail

### PR DESCRIPTION
**Pull request by YuvHayun**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
added credentials for the *ID or Client ID* and *Token or Tenant ID* and relevant tests.

## Does it break backward compatibility?
   - [ ] Yes
   - [X] No

### CODE REVIEW

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 1]
```diff
@@ -1594,8 +1594,10 @@ def main():
     args: dict = demisto.args()
     params: dict = demisto.params()
     self_deployed: bool = params.get('self_deployed', False)
-    tenant_id: str = params.get('tenant_id', '') or params.get('_tenant_id', '')
-    auth_and_token_url: str = params.get('auth_id', '') or params.get('_auth_id', '')
+    tenant_id: str = params.get('tenant_id', '') or params.get('_tenant_id', '') or (params.get('creds_tenant_id')
```
Comment (ShahafBenYakir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:1597]: Let's add a comment here saying why we have these different param names (due to the credentials change)

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail_test.py [hunk 2]
```diff
@@ -6,11 +6,34 @@
...
+])
+def test_params(mocker, params, expected_result):
+    """
+    Given:
+      - Configuration parameters
+    When:
+      - One of the required parameters are missed.
+    Then:
+      - Ensure the exception message as expected.
+    """
```
Comment (ShahafBenYakir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail_test.py:27]: Added cases: `Case 1: ...` (See NeiscoutSigtline for example)

Comment (ShahafBenYakir) [reply Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail_test.py:27]: Also let's add a test to show that no matter which of the three options is used by the user the relevant params are wprking.

**Review by ShahafBenYakir — CHANGES_REQUESTED** [review]

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/pipfile [hunk 3]
```diff
@@ -19,7 +19,7 @@ networkx = "==2.5"
 beautifulsoup4 = "==4.9.3"
 freezegun = "==0.3.12"
 vulture = "==1.5"
-parinx = {ref = "6493798ceba8089345d970f71be4a896eb6b081d", git = "https://github.com/coyo8/parinx.git"}
+parinx = {ref = "6493798ceba8089345d970f71be4a896eb6b081d", git = "https://github.com/coyo8/parinx.git", egg = "parinx"}
```
Comment (ShahafBenYakir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/pipfile:22]: What is this change?

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail_test.py [hunk 4]
```diff
@@ -6,11 +6,70 @@
...
+    assert expected_result in str(e.value)
+
+
+@pytest.mark.parametrize('params, expected_results', [
+    ({'creds_tenant_id': {'password': '1234'}, 'creds_auth_id': {'password': '3124'},
+      'credentials': {'password': '2412'}}, ['1234', '3124', '2412']),
+    ({'tenant_id': '5678', 'enc_key': '8142', 'auth_id': '5678'}, ['5678', '5678', '8142']),
+    ({'_tenant_id': '1267', 'credentials': {'password': '1234'}, '_auth_id': '8888'}, ['1267', '8888', '1234'])
+])
+def test_params_working(mocker, params, expected_results):
```
Comment (ShahafBenYakir) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail_test.py:48]: Nice!

**Review by ShahafBenYakir — APPROVED** [review]
See my comment

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/pipfile (hunk 3 above)
Comment (YuvHayun) [reply Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/pipfile:22]: Something I needed for the testing, i'll revert it

**Pull request merged**